### PR TITLE
Update eth wallet to use new client

### PR DESCRIPTION
**Pull request by krhubert**
~~Waiting for:~~

~~https://github.com/mesg-foundation/engine/pull/1146~~

Client has:

- access to all apis via grpc
- (Un)Marshal functions to not exposed internal input format
- execution runner that is a nice wrapper that could be used by services 

TODO:

- [ ] Test and find a better name for ExecutionRunner and ExeuctionRun.

Problems:

The client has to support both inputs and errors (same for the services). Now service only support valid inputs, but with workflow, an error output must be supported as well. We need to think of good client api for that :

```go
func (input []byte, errOut error) ([]byte, error)
```

I don't like it, maybe some type, or enum, don't know.

### CODE REVIEW

File: client/execution_runner.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func (r *ExecutionRunner) Add(name string, fn ExecutionRun) {
+	r.defs[name] = fn
+}
+
+func (r *ExecutionRunner) Run() error {
+	stream, err := r.client.ExecutionClient.Stream(context.Background(), &pb.StreamExecutionRequest{
+		Filter: &pb.StreamExecutionRequest_Filter{
+			InstanceHash: r.client.InstanceHash,
```
Comment (antho1404) [inline client/execution_runner.go]: as a service, you should only listen to execution with the status in-progress

**Comment by NicolasMahe** [comment]
> * [ ]  Test and find a better name for ExecutionRunner and ExeuctionRun.

typo

**Comment by krhubert** [comment]
close #1157

**Comment by NicolasMahe** [comment]
Tested with CLI 👍

**Pull request merged**